### PR TITLE
gRPC for cross DC traffic

### PR DESCRIPTION
**Pull request by vytautas-karpavicius**
**What changed?**
- Updated `DispatcherProvider` interface to support creating dispatchers for gRPC. Updated `dnsDispatcherProvider` implementation to accordingly.
- Client factory will instantiate either Thrift or GRPC client based on the given dispatcher. These clients are used for:
  - admin client: for cross dc replication
  - frontend client: for request forwarding
- Updated config section within `ClusterGroupMetadata` to allow configuring cross DC transport. If not specified, it defaults to `tchannel`, as currently used.

**Why?**
To support gRPC for cross DC traffic.

**How did you test it?**
Integration tests for cross DC replication.

**Potential risks**
TChannel is still used by default, and gRPC is currently opt in to minimize risk.

**Release notes**
Updated CHANGELOG.md

**Documentation Changes**

### CODE REVIEW

File: client/clientBean.go [hunk 1]
```diff
@@ -65,9 +66,10 @@ type (
 		SetRemoteFrontendClient(cluster string, client frontend.Client)
 	}
 
-	// DispatcherProvider provides a diapatcher to a given address
+	// DispatcherProvider provides a dispatcher to a given address
 	DispatcherProvider interface {
 		Get(name string, address string) (*yarpc.Dispatcher, error)
```
Comment (Shaddoll) [inline client/clientBean.go:71]: Can you rename it to GetTchannel to make it explicit?

File: client/clientfactory.go [hunk 2]
```diff
@@ -358,3 +374,8 @@ func (cf *rpcClientFactory) newFrontendGRPCClient(hostAddress string) (frontend.
 		apiv1.NewVisibilityAPIYARPCClient(config),
 	), nil
 }
+
+func isGRPCOutbound(config transport.ClientConfig) bool {
+	namer, ok := config.GetUnaryOutbound().(transport.Namer)
```
Comment (Shaddoll) [inline client/clientfactory.go:379]: should we panic/log.fatal here if `ok==false`?

**Review by Shaddoll — APPROVED** [review]

File: client/clientBean.go (hunk 1 above)
Comment (vytautas-karpavicius) [reply client/clientBean.go]: Yes, updated.

File: client/clientfactory.go (hunk 2 above)
Comment (vytautas-karpavicius) [reply client/clientfactory.go:379]: Unless old version of yarpc is used, this will not happen. But I guess explicit early panic is better. Updated.

**Pull request merged**